File: dehaze-go/internal/model/sys_algorithm_favorite.go
package model

// SysAlgorithmFavorite 算法收藏表
type SysAlgorithmFavorite struct {
	BaseModel
	UserID      int64 `gorm:"column:user_id;type:bigint;not null;uniqueIndex:uk_user_algorithm;comment:用户ID" json:"userId"`
	AlgorithmID int64 `gorm:"column:algorithm_id;type:bigint;not null;uniqueIndex:uk_user_algorithm;comment:算法ID" json:"algorithmId"`
	Deleted     int64 `gorm:"column:deleted;type:bigint;not null;default:0;comment:逻辑删除标识(0:未删除;>0:已删除,值为删除时的行id)" json:"deleted"`
}

func (SysAlgorithmFavorite) TableName() string {
	return "sys_algorithm_favorite"
}

File: dehaze-python/app/models/entity/sys_algorithm_favorite.py
"""
算法收藏实体
"""
from sqlalchemy import BigInteger, Index, DateTime, UniqueConstraint
from datetime import datetime, timezone
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base


class SysAlgorithmFavorite(Base):
    """算法收藏表"""
    __tablename__ = 'sys_algorithm_favorite'
    __table_args__ = (
        UniqueConstraint('user_id', 'algorithm_id', name='uk_user_algorithm'),
        Index('idx_user_id', 'user_id'),
        Index('idx_algorithm_id', 'algorithm_id'),
        {'comment': '算法收藏表'},
    )

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True, autoincrement=True, comment='id')
    user_id: Mapped[int] = mapped_column(BigInteger, nullable=False, comment='用户ID')
    algorithm_id: Mapped[int] = mapped_column(BigInteger, nullable=False, comment='算法ID')
    create_time: Mapped[datetime] = mapped_column(
        DateTime, nullable=False, default=lambda: datetime.now(timezone.utc), comment='收藏时间'
    )
